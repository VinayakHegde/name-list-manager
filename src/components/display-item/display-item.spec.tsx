import * as React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import DisplayItem from ".";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const withProps = (props) => {
  return <DisplayItem {...props} />;
};

describe("<DisplayItem />", () => {
  const props = {
    name: "Hello",
    onClick: jest.fn(),
    btnName: "Test",
  };
  it("should match the snapshot", async () => {
    act(() => {
      render(withProps(props), container);
    });
    expect(container).toMatchSnapshot();
  });
  it("should render elements", async () => {
    act(() => {
      render(withProps(props), container);
    });
    const name = container.querySelector(".displayItem__name");
    expect(name.innerText).toMatchSnapshot(props.name);
    const btnName = container.querySelector(".displayItem__name");
    expect(btnName.innerText).toMatchSnapshot(props.btnName);
  });
});
