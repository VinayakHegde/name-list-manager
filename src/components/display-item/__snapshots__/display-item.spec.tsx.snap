// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisplayItem /> should match the snapshot 1`] = `
<div>
  <div
    class="displayItem"
  >
    <span
      class="displayItem__name"
    >
      Hello
    </span>
    <button
      class="primary"
    >
      Test
    </button>
  </div>
</div>
`;

exports[`<DisplayItem /> should render elements: Hello 1`] = `undefined`;

exports[`<DisplayItem /> should render elements: Test 1`] = `undefined`;
